src/main/kotlin/org/rust/lang/core/resolve/ref/RsMarcoItemReferenceContributor.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.*
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+
+class RsMarcoItemReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        val pattern = PlatformPatterns.psiElement(IDENTIFIER)//never called
```

Comment:
Reference providers are only called for elements implementing `ContributedReferenceHost` and I don't think we need to use it here. 

Contributed references are useful if you want to extend go to definition for some language *from outside*. For example, if you are creating support for a particular framework, which renders template like `ctx.render("template_file_name.ctx")` , you can *inject* reference to a file into the string literal, provided that string literal in the language is a `ContributedReferenceHost`. 

Here, we have control over PSI, so we can return the reference from `getReference` directly.